### PR TITLE
2159 work package templates hook up to createedit wp pages

### LINKED ISSUE #2159: Work Package Templates - Hook up to create/edit WP pages
### Description

Blocked By #2148 #2156 

We want the work package templates to be accessible from the create/edit work packages pages. This ticket should be done after the get all work package template endpoint is done. See the mocks below for how the templates should be displayed. When a template is selected it should fill in all the corresponding fields. For blocked by it should search for a work package on the project associated with the CR that has the corresponding stage and name saved in the work package template's blocked by info (gonna be tricky)

### Acceptance Criteria

- [ ] When you click on a work package template it should fill in all the corresponding fields on the work package form
- [ ] If you're using it on an edit, it should fill around the existing values in the form (if an input already has a value in the input, it won't replace it)

### Proposed Solution

This one's gonna be a lot of react-hook-form trickiness, reference the docs for help, and ask a lot of questions!: https://react-hook-form.com/

### Mocks
image

### PR DESCRIPTION
**Pull request by AShyamkumar19**
## Changes

So far, created a new file called WorkPackageTemplateDetails where it's responsible for mapping out all the boxes in the WorkPackageTemplates. This react component is called in the WorkPackageView. Once the user clicks on the box template, the respective text fields are field with information from the template. 

## Screenshots

Screenshot 2024-06-11 at 11 00 46 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #2159

### CODE REVIEW

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Your tests are failing you may have to merge develop into your branch

**Review by superhvarn — CHANGES_REQUESTED** [review]
Remove the project hook changes

**Review by superhvarn — DISMISSED** [review]

File: src/frontend/src/pages/WorkPackageForm/WorkPackageTemplateDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+export const WorkPackageTemplateSection: React.FC<WorkPackageTemplateProps> = ({
+  workPackageTemplates,
+  currentWorkPackageTemplate,
+  setCurrentWorkPackageTemplate
+}: WorkPackageTemplateProps) => {
+  return (
+    <>
+      <Typography variant="h5">Templates</Typography>
+      <Box sx={{ display: 'flex', flexDirection: 'row' }}>
```
Comment (walker-sean) [inline src/frontend/src/pages/WorkPackageForm/WorkPackageTemplateDetails.tsx]: See if you can make this component cause the cursor to go into "click mode"

File: src/frontend/src/pages/WorkPackageForm/WorkPackageTemplateDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { WorkPackageTemplate } from 'shared';
+import { Box, Typography } from '@mui/material';
+
+interface WorkPackageTemplateProps {
+  workPackageTemplates: WorkPackageTemplate[];
+  currentWorkPackageTemplate?: WorkPackageTemplate;
+  setCurrentWorkPackageTemplate: (value: WorkPackageTemplate) => void;
+}
+
+export const WorkPackageTemplateSection: React.FC<WorkPackageTemplateProps> = ({
```
Comment (walker-sean) [inline src/frontend/src/pages/WorkPackageForm/WorkPackageTemplateDetails.tsx]: Rename the file to match the component name

**Review by walker-sean — CHANGES_REQUESTED** [review]
This is looking really good so far!

A few things we're missing:
- We need to also apply the blocked by information (you can look at the ticket to see the logic that goes into that
- We need to check whether a user has input new data into the field. If they have, do not overwrite. This can be done by comparing the value to default value (empty or an actual value if its an edit)

**Review by superhvarn — CHANGES_REQUESTED** [review]
lgtm just make sure to keep the text within the work package template boxes when collapsing the screen

File: src/frontend/src/pages/WorkPackageForm/WorkPackageTemplateSection.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      <Box sx={{ display: 'flex', flexWrap: 'wrap', cursor: 'pointer' }}>
+        {workPackageTemplates.map((template) => (
+          <Box
+            key={template.workPackageTemplateId}
+            onClick={() => {
+              setCurrentWorkPackageTemplate(template);
+            }}
+            sx={{
+              width: { xs: '10em', sm: '12em', md: '15em', lg: '18em' },
+              height: { xs: '4em', sm: '5em', md: '6em', lg: '7em' },
```
Comment (Aaryan1203) [inline src/frontend/src/pages/WorkPackageForm/WorkPackageTemplateSection.tsx]: See if there is a better way to do this for the styling, this isn't good practice

Comment (superhvarn) [reply src/frontend/src/pages/WorkPackageForm/WorkPackageTemplateSection.tsx]: I concur

**Review by superhvarn — APPROVED** [review]
lgtm

**Review by walker-sean — APPROVED** [review]
LGTM

**Pull request merged**